.github/workflows/check-links.yaml

```diff
@@ -0,0 +1,42 @@
+name: docs
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  link-checker:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
+    steps:
+      - uses: actions/checkout@v2
+      
+      # Using markdown-link-check to check markdown docs
+      - uses: gaurav-nelson/github-action-markdown-link-check@v1
+        with:
+          use-verbose-mode: 'yes'
+          config-file: '.github/workflows/markdown-link-check-config.json'
+          max-depth: 1
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r userdocs/requirements.txt
+      - name: Build docs for link check
+        run: cd userdocs; mkdocs build
```

Comment:
any reason why we are not using _make target_ here (e.g. make build-pages) ? Just to avoid duplicating if there is any change.

Reply:
I changed this, but as you can see in https://github.com/weaveworks/eksctl/pull/2062/checks?check_run_id=607591568 this will download a bunch of go modules. :shrug: